File: app/collections/WikipediaCollections.js
module.exports = function(dbInstance) {

	var collections = {
		events: dbInstance.collection('EventsCollection'),
		births: dbInstance.collection('BirthsCollection'),
		deaths: dbInstance.collection('DeathsCollection'),
		holidaysandobservances: dbInstance.collection(
			'HolidaysAndObservancesCollection'
		)
	};

	collections['events'].createIndex({title: 1}, {unique: true})
		.then(function(index) {
			console.log('Index Events:', index);
		})
		.catch(function(err){
			console.log(arguments);
		});
	collections['births'].createIndex({title: 1}, {unique: true})
		.then(function(index){
			console.log('Index Events:', index);
		})
		.catch(function(err){
			console.log(arguments);
		});
	collections['deaths'].createIndex({title: 1}, {unique: true})
		.then(function(index){
			console.log('Index Events:', index);
		})
		.catch(function(err){
			console.log(arguments);
		});
	collections['holidaysandobservances'].createIndex({title: 1}, {unique: true})
		.then(function(index){
			console.log('Index Events:', index);
		})
		.catch(function(err){
			console.log(arguments);
		});

	return collections;
}

File: python/app/collections/WikipediaCollections.py
class Singleton(type):
    _instances = {}
    def __call__(cls, *args, **kwargs):
        if cls not in cls._instances:
            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
        return cls._instances[cls]

class WikipediaCollections():
    __metaclass__ = Singleton
    def __init__(self, dbConnection):
        if dbConnection and dbConnection['connect']:
            self.collections = {
                'events' : dbConnection['EventsCollection'],
                'births' : dbConnection['BirthsCollection'],
                'deaths' : dbConnection['DeathsCollection'],
                'holidaysandobservances' : dbConnection['Holidaysandobservances']
            }
        else:
            self.collections = {}

    def getCollections(self):
        return self.collections

